qiskit/circuit/quantumcircuit.py

```diff
@@ -681,7 +681,12 @@ def compose(self, other, qubits=None, clbits=None, front=False, inplace=False):
 
         for instr, _, _ in mapped_instrs:
             dest._update_parameter_table(instr)
-        dest._calibrations.update(other.calibrations)
+
+        for gate, cals in other.calibrations.items():
+            if gate in dest._calibrations:
+                dest._calibrations[gate].update(cals)
+            else:
+                dest._calibrations[gate] = cals
```

Comment:
There's actually another bug I meant to fix:
```
--- a/qiskit/converters/dag_to_circuit.py
+++ b/qiskit/converters/dag_to_circuit.py
@@ -11,6 +11,8 @@
 # that they have been altered from the originals.

 """Helper function for converting a dag to a circuit."""
+from collections import defaultdict
+
 from qiskit.circuit.quantumcircuit import QuantumCircuit


@@ -48,7 +50,8 @@ def dag_to_circuit(dag):
     name = dag.name or None
     circuit = QuantumCircuit(*dag.qregs.values(), *dag.cregs.values(), name=name,
                              global_phase=dag.global_phase)
-    circuit.calibrations = dag.calibrations
+    circuit._calibrations = defaultdict(dict)
+    circuit._calibrations.update(dag.calibrations)

     for node in dag.topological_op_nodes():
         # Get arguments for classical control (if any)
```

After fixing that, this should be:
```suggestion
        for gate, cals in other.calibrations.items():
            dest._calibrations[gate].update(cals)
```

Reply:
Done.